templates/influxdb.conf.epp

```diff
@@ -12,15 +12,18 @@
       Array $master_list
 | -%>
 
-<% unless $master_list.empty { 
-      $serverarray = $master_list.map |$host| { 
-          $server_tag = join(split($host,"[.]").map | $element | { "server" },'.') 
-          $wildcards = join(split($host,"[.]").map | $element | { "*" },'.') 
-          "puppetlabs.${wildcards}.placeholder .${server_tag}.measurement*" 
-     }
-   }
+<%
+$serverarray = $master_list.map |$host| { 
+  $server_tag = join(split($host,"[.]").map | $element | { "server" },'.') 
+  $wildcards = join(split($host,"[.]").map | $element | { "*" },'.') 
+  "puppetlabs.${wildcards}.placeholder .${server_tag}.measurement*" 
+}
 
-$unique_serverarray = unique($serverarray) 
+if length(delete_undef_values($serverarray)) > 0 {
```

Comment:
It looks like you don't actually need this (but it's not hurting anything). I did a test:

~~~
❯ puppet apply -e 'notice(unique([]).to_json())'
Notice: Scope(Class[main]): []
❯ puppet apply -e 'notice(unique([undef,undef]).to_json())'
Notice: Scope(Class[main]): [null]
~~~

(I trimmed some noise from the puppet run.)